### PR TITLE
push release from tags

### PR DESCRIPTION
**Pull request by giorgiosironi**

### CODE REVIEW

**Comment by giorgiosironi** [comment]
https://github.com/elifesciences/sciencebeam/blob/develop/sciencebeam/__init__.py has the hardcoded version. `setup.py` depends on that. What purpose does it serve?

File: Jenkinsfile [hunk 1]
```diff
@@ -44,26 +31,19 @@ elifePipeline {
...
-            stage 'Push release image', {
-                if (isNew) {
-                    def image = DockerImage.elifesciences(this, 'sciencebeam', commit)
-                    image.tag('latest').push()
-                    image.tag(candidateVersion).push()
-                }
-            }
+        }
+
+        elifeTagOnly { tagName ->
```
Comment (giorgiosironi) [inline Jenkinsfile]: this is executed in response to `git tag v0.0.2 && git push origin v0.0.2` but not in other cases

**Comment by giorgiosironi** [comment]
Still `[wip]` due to the need to remove `sciencebeam.__version__`.

**Comment by de-code** [comment]
> Still `[wip]` due to the need to remove `sciencebeam.__version__`.

I guess that will be needed for the pypi release (which we currently don't have for this project). But it's probably of little use within the docker container. It could be of some use when installed from git but then we would need to set the version before creating the tag and we decided against that. It doesn't seem to be worth it.

File: Jenkinsfile [hunk 2]
```diff
@@ -10,18 +8,16 @@ elifePipeline {
...
-            ).trim()
-            echo "Candidate version: ${candidateVersion}"
-            isNew = sh(script: "git tag | grep v${candidateVersion}", returnStatus: true) != 0
-            echo "isNew: ${isNew}"
+            // TODO: extract into dockerComposeBuild
+            def version
+            if (env.TAG_NAME) {
+                version = env.TAG_NAME - 'v'
+            } else {
+                version = 'develop'
```
Comment (de-code) [inline Jenkinsfile]: I wonder whether it should instead be blank? It's not a semantic version and pypi wouldn't accept that.

Comment (giorgiosironi) [reply Jenkinsfile]: Good for me, the Docker image label won't be very clear but such an image should never be pushed.

File: Jenkinsfile [hunk 3]
```diff
@@ -10,18 +8,16 @@ elifePipeline {
...
-        stage 'Check version', {
-            candidateVersion = dockerComposeRunAndCaptureOutput(
-                "sciencebeam",
-                "./print_version.sh",
-                commit
-            ).trim()
-            echo "Candidate version: ${candidateVersion}"
-            isNew = sh(script: "git tag | grep v${candidateVersion}", returnStatus: true) != 0
-            echo "isNew: ${isNew}"
+            // TODO: extract into dockerComposeBuild
```
Comment (de-code) [inline Jenkinsfile]: Are we waiting for extracting this before it can go in?

**Comment by de-code** [comment]
(I would like to release a new version very soon)

File: Jenkinsfile (hunk 3 above)
Comment (giorgiosironi) [reply Jenkinsfile]: not a blocker, shouldn't hold this

Comment (de-code) [reply Jenkinsfile]: Okay, any other objections against merging it now?

**Comment by giorgiosironi** [comment]
The missing thing was that the version is known to the code:
https://github.com/elifesciences/sciencebeam/blob/develop/setup.py#L79
whereas it should only be in the tag.

**Comment by de-code** [comment]
> The missing thing was that the version is known to the code:
> https://github.com/elifesciences/sciencebeam/blob/develop/setup.py#L79
> whereas it should only be in the tag.

Hmm okay, I guess we should fix that. Good point. I guess we should fix that first. We did in [sciencebeam-utils](https://github.com/elifesciences/sciencebeam-utils). This `sciencebeam` project isn't actually pushing to pypi (at least not currently). The `setup.py` file would be used when doing a bulk conversion via Apache Beam - it's bundling up the source and installs it from that bundle where it wants to run it (e.g. Google Dataflow - using Google Storage intermediately).

In any case we should use the set version script at build time as for `sciencebeam-utils`. Not sure what Apache Beam will do with a version of develop or blank though..

**Comment by giorgiosironi** [comment]
It can be merged as-is with the danger that the version number should always be updated in `master` before a tag can be made. It's easy to forget.

**Comment by de-code** [comment]
> It can be merged as-is with the danger that the version number should always be updated in `master` before a tag can be made. It's easy to forget.

Better to fix it first. Will copy and refine scripts from `sciencebeam-utls`.

**Comment by de-code** [comment]
If a "version" (from the tag) was provided, it will use that. Otherwise it will generate the version. It's generating valid versions using the date and commit like in `sciencebeam-utils`. But it's now doing it at build time. e.g. `2019.6.7.dev228940219`.

**Comment by de-code** [comment]
(this might still cause issues when running Apache Beam locally - but could look into that when needed)

**Comment by de-code** [comment]
Merging now. One thing that could be improved is to pass in the date to allow for consistent build. Although it should do for now.

**Pull request merged**